transport.go

```diff
@@ -130,9 +139,14 @@ func (t *HTTPTransport) Configure(options ClientOptions) {
 		Logger.Printf("%v\n", err)
 		return
 	}
-
 	t.dsn = dsn
-	t.buffer = make(chan *http.Request, t.BufferSize)
+
+	t.buffer = make(chan batch, 1)
```

Comment:
The buffered channel with capacity 1 works as a mutex, ensuring only one goroutine can access the current batch at a given time.